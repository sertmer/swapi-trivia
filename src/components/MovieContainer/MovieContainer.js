import React from 'react';
import './MovieContainer.scss';
import MovieCard from '../MovieCard/MovieCard';

const MovieContainer = ({ movies, updateState }) => {
  let movieCards = movies.map(movie => {
    return (
      <MovieCard 
        className='MovieCard'
        key={movie.episode_id}
        episode={movie.episode_id}
        title={movie.title}
        releaseDate={movie.release_date}
        get
        updateState={updateState}
      />
    ) 
  })
  return (
    <main>
      {movieCards}
    </main>
  )
}

export default MovieContainer