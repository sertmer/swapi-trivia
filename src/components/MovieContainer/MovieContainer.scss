/* movie container test */
main {
  display: flex;
  flex-flow: row wrap;
}