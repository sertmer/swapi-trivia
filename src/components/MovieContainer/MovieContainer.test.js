import React from 'react'
import MovieContainer from './MovieContainer'
import { shallow } from 'enzyme'

describe('MovieContainer', () => {
  let wrapper
  let mockMovies = {
    episode_id: '1',
    title: 'spaceballs',
    release_date: '12-12-12',
  }

  let mockUpdateState = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
    <MovieContainer 
      movies={[mockMovies]}
      updateState={mockUpdateState}  
    />)
  })

  it('should match the snapshot', () => {   
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the loading snapshot', () => {
    wrapper = shallow(
      <MovieContainer 
        movies={[]}
        updateState={mockUpdateState}
      />)
    expect(wrapper).toMatchSnapshot()
  })
})