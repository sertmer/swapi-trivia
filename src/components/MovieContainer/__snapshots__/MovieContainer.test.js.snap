// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the loading snapshot 1`] = `<Loading />`;

exports[`MovieContainer should match the snapshot 1`] = `
<main>
  <MovieCard
    className="MovieCard"
    episode="1"
    key="1"
    releaseDate="12-12-12"
    title="spaceballs"
    updateState={[MockFunction]}
  />
</main>
`;
